### ISSUE TITLE
Clarification on Bilinear Attention

### ISSUE THREAD

**Issue by s-j-99** [hongli-ma/RNAsmol#2]
Hi, I was looking through your model and I had a question about the bilinear attention (BAN) layer implemented in the final model. It outputs attention maps and logits, but neither of these are used during model training to compute an output for the loss function. So how exactly would this layer be trained or contribute to the final classification performance? Are the input RNA and ligand embeddings transformed by this layer? Because it looks more like the embeddings are copied and then attention is computed.

**Comment by hongli-ma** [comment]
Hi, BAN can be trained by replacing 'score = self.classifier(x)' with  'score = self.classifier(f)' in class MCNN_GCN.

**Comment by BHM-Bob** [comment]
> Hi, BAN can be trained by replacing 'score = self.classifier(x)' with 'score = self.classifier(f)' in class MCNN_GCN.

BUT, the shape of `f` may be the `[1, filter_num * 3 * 2]`, whose batch dim is 1!

## code
```python
# append after model.py
if __name__ == "__main__":
    filter_num, embedding_size, ban_heads = 32, 96, 1
    model = BANLayer(v_dim=filter_num*3, q_dim=embedding_size, h_dim=filter_num*3*2, h_out=ban_heads)
    print(model)
    
    batch_size = 128
    v = torch.randn(batch_size, filter_num*3)
    q = torch.randn(batch_size, embedding_size)
    f, att = model(v, q)
    print(f.shape)
    print(att.shape)
```
outputs:
```text
BANLayer(
  (v_net): FCNet(
    (main): Sequential(
      (0): Dropout(p=0.2, inplace=False)
      (1): Linear(in_features=96, out_features=576, bias=True)
      (2): ReLU()
    )
  )
  (q_net): FCNet(
    (main): Sequential(
      (0): Dropout(p=0.2, inplace=False)
      (1): Linear(in_features=96, out_features=576, bias=True)
      (2): ReLU()
    )
  )
  (p_net): AvgPool1d(kernel_size=(3,), stride=(3,), padding=(0,))
)
torch.Size([1, 192])
torch.Size([1, 1, 128, 128])
```
